bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/VehicleHandler.java

```diff
@@ -0,0 +1,576 @@
+                    getImageResources();
+                }
+                String key = command.toFullString() + "_" + config.get().vin;
+                String encodedImage = EMPTY;
+                if (imageStorage.get().containsKey(key)) {
+                    encodedImage = imageStorage.get().get(key);
+                    logger.trace("Image {} found in storage", key);
+                } else {
+                    logger.trace("Request Image {} ", key);
+                    encodedImage = getImage(command.toFullString());
+                    if (!encodedImage.isEmpty()) {
+                        imageStorage.get().put(key, encodedImage);
+                    }
+                }
+                if (encodedImage != null && !encodedImage.isEmpty()) {
+                    RawType image = new RawType(Base64.getDecoder().decode(encodedImage),
+                            MIME_PREFIX + config.get().format);
+                    updateState(new ChannelUID(thing.getUID(), GROUP_IMAGE, "image-data"), image);
+                } else {
+                    logger.debug("Image {} is empty", key);
+                }
+            }
+        } else if (channelUID.getIdWithoutGroup().equals("clear-cache") && command.equals(OnOffType.ON)) {
+            List<String> removals = new ArrayList<String>();
+            imageStorage.get().getKeys().forEach(entry -> {
+                if (entry.contains("_" + config.get().vin)) {
+                    removals.add(entry);
+                }
+            });
+            removals.forEach(entry -> {
+                imageStorage.get().remove(entry);
+            });
+            updateState(new ChannelUID(thing.getUID(), GROUP_IMAGE, "clear-cache"), OnOffType.OFF);
+            getImageResources();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = Optional.of(getConfigAs(VehicleConfiguration.class));
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            updateStatus(ThingStatus.UNKNOWN);
+            BridgeHandler handler = bridge.getHandler();
+            if (handler != null) {
+                accountHandler = Optional.of((AccountHandler) handler);
+                startSchedule(config.get().refreshInterval);
+                if (!config.get().vin.equals(NOT_SET)) {
+                    imageStorage = Optional.of(storageService.getStorage(BINDING_ID + "_" + config.get().vin));
+                    if (!imageStorage.get().containsKey(EXT_IMG_RES + config.get().vin)) {
+                        getImageResources();
+                    }
+                    setImageOtions();
+                }
+                updateState(new ChannelUID(thing.getUID(), GROUP_IMAGE, "clear-cache"), OnOffType.OFF);
+            } else {
+                String textKey = Constants.PREFIX + "vehicle" + Constants.STATUS_BRIDGEHANDLER_MISSING;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_MISSING_ERROR, textKey);
```

Comment:
I never observed this case so it's a bit hard for me to report the proper state. 
Adding `RuntimeException` reports a medium checkstyle issue because general exceptions shall not be used.
Having a `Bridge` not null but no handler attached shall not happen in principle, right?

According to the status table this shall report the proper status 
```
updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, textKey);
```

Reply:
> Adding `RuntimeException` reports a medium checkstyle issue because general exceptions shall not be used.

Sorry for being unclear, with "runtime exception" I meant some appropriate subclass of `RuntimeException`. It could even be a custom defined one, but for this case it might be overkill. Maybe `IllegalStateException` could be suitable? Basically we are dealing with null annotations here, because otherwise we could simply do nothing, as we know that `NullReferenceException` won't happen.

> Having a `Bridge` not null but no handler attached shall not happen in principle, right?

Exactly.

> According to the status table this shall report the proper status
> 
> ```
> updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, textKey);
> ```

Yes, but this should not be set manually - the framework takes care of this status. So the point is that framework won't call `initialize()` until parent thing (bridge) is already initialized (i.e. online, offline or unknown).

Reply:
Ok, `llegalStateException` can be thrown without checkstyle complaining.
Just to get it straight for me:
Looked into `BaseThingHandler` and it doesn't handle Exceptions. So

1. I'm sure some kind of _ThreadPool_ is handling this without breaking the thread forever?
2. The `VehicleHandler` status will stuck at `Ìnitializing`?

Reply:
I'm not sure exactly what would happen, but you could try throwing one in order to find out. However, the point is that it can't happen. See also #12331.